### PR TITLE
Fixes #5233 | Replace NULL by nullptr

### LINKED ISSUE #5233: Replace NULL by nullptr

### PR DESCRIPTION
**Pull request by janisz**
Done by running: `run-clang-tidy.py src -checks=-*,modernize-use-nullptr
-fix`

### CODE REVIEW

**Comment by danielmewes** [comment]
Thanks @janisz !
If you haven't done so yet, would you mind taking a look at http://rethinkdb.com/community/cla/ and signing the CLA? We require this for legal reasons before we can merge a pull request.

The change itself looks good as far as I've seen in a quick glance.

I personally would prefer not to apply this change to the files in `src/rapidjson`, `src/cjson` and in `rdb_protocol/geo/s2` since those are based on external projects.
The further we diverge from the upstream code, the more difficult it becomes to rebase and/or backport upstream fixes from those projects to our code in the future.

Others might have different opinions on this though.
@VeXocide what do you think?

**Comment by mlucy** [comment]
+1 for not applying the cleanup to external projects.

**Comment by janisz** [comment]
I signed CLA. I reverted changes to external projects. I see `calang-tidy` do not replace all occurrences of `NULL`. It skip `NULL`s used in conditions (probably because it's not 100% safe to replace), should I replace them too?

**Comment by VeXocide** [comment]
I agree we shouldn't apply these changes to external projects, for the reason mentioned.

**Comment by danielmewes** [comment]
@janisz I think in our code base it should be safe to replace all `NULL`s by `nullptr`, even in conditions. If you have time to do it, that would be great. Otherwise the current state of this PR is already a good step forward.

**Comment by janisz** [comment]
I replaced NULLs with `sed`, it compiles but I have problem with tests. I run `make test` and

> 74 of 165 tests failed

most of them failed with 

> [Errno 2] No such file or directory

How should I run those tests?

**Comment by danielmewes** [comment]
@janisz For running the tests and get somewhat better output, you can try:
```
$ make test-deps
$ cd build/release
$ ./rethinkdb-unittest
```

There are also integration tests:
```
$ cd test/rql_test
$ ./test-runner -j 9
```

If the tests don't work for you, please let me know. The tests might require some additional dependencies to work properly. I can simply check out your branch and run the tests on our system.

**Comment by janisz** [comment]
Thanks, I couldn't find it in docs. I run unittests and one of them failed but when I run without others it pass. Probably some issues with my setup.
```
$ ./rethinkdb-unittest
...
[==========] 332 tests from 70 test cases ran. (237019 ms total)
[  PASSED  ] 331 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] JSProc.Passthrough

$ ./rethinkdb-unittest --gtest_filter=JSProc.Passthrough
Note: Google Test filter = JSProc.Passthrough
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from JSProc
[ RUN      ] JSProc.Passthrough
[       OK ] JSProc.Passthrough (490 ms)
[----------] 1 test from JSProc (490 ms total)
[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (490 ms total)
[  PASSED  ] 1 test.
```

**Comment by danielmewes** [comment]
Yeah, the `JSProc.Passthrough` test is known to fail intermittently in some cases (https://github.com/rethinkdb/rethinkdb/issues/5065). So that looks good.

We'll make a final pass over your changes and should be ready to merge once we're done.

Thanks again for your work on this!

File: src/clustering/administration/issues/non_transitive.cc [hunk 1]
```diff
@@ -73,5 +73,5 @@ std::vector<scoped_ptr_t<issue_t> > non_transitive_issue_tracker_t::get_issues(
     if (interruptor->is_pulsed()) {
         throw interrupted_exc_t();
     }
-    return std::move(issues);
```
Comment (danielmewes) [inline src/clustering/administration/issues/non_transitive.cc]: Did you make this change? (it seems correct, just wondering)

**Comment by danielmewes** [comment]
It looks good. Merging.
Thanks again @janisz .

File: src/clustering/administration/issues/non_transitive.cc (hunk 1 above)
Comment (janisz) [inline src/clustering/administration/issues/non_transitive.cc]: Definitely not. I just run `run-clang-tidy.py src -checks=-*,modernize-use-nullptr -fix`

**Pull request merged**